frontend/src/pages/HomePage/index.tsx

```diff
@@ -3,12 +3,40 @@ import { Link, useHistory } from "react-router-dom";
 import Button from "../../components/Button";
 import TextField from "../../components/TextField";
 import styles from "./style.module.css";
+import validateGame from "../../api/validateGame";
+import createGame from "../../api/createGame";
 
 const HomePage = () => {
   const [gameCode, setGameCode] = useState("");
+  const [gameCodeError, setGameCodeError] = useState("");
+  const [newGameError, setNewGameError] = useState("");
 
   const browserHistory = useHistory();
 
+  const handleJoinGame = async () => {
+    const res = await validateGame({ gameCode });
+
+    if (res.success) {
+      browserHistory.push(`/${gameCode}`);
+    } else {
+      setGameCodeError(res.error);
+    }
+  };
+
+  const handleNewGame = async () => {
+    const res = await createGame();
+
+    if (res.success) {
+      if (res.data) {
+        browserHistory.push(`/${res.data}`);
+      } else {
+        setNewGameError("Unknown Error, please try again");
+      }
+    } else {
+      setNewGameError(res.error);
```

Comment:
This returns the following error:

<img width="1047" alt="image" src="https://user-images.githubusercontent.com/32445605/117100217-515bcd00-adc7-11eb-9cab-ef482d6c453b.png">

Reply:
Ah yeah this is a problem with createGame(), it's returning an object as res.error, which this then tries to render

Reply:
Interesting - is this expected behaviour @Alxjlv?

Reply:
I should just be able to return a generic server error message, I don't think there's anything useful in the XMLHttpRequest object to return to the user

Reply:
Do you still get the error now @taitfuller?

Reply:
This is fixed now - great work!